### PR TITLE
WRO-630: Add support for SCSS/Sass

### PR DESCRIPTION
**Pull request by hong6316**
### Issue Resolved / Feature Added
We would like to add support for SCSS to Enact CLI

### Resolution
* Add support for modules SCSS and Sass
* Add documentation.

### Links
WRO-630

### Comments
Enact-DCO-1.0-Signed-off-by: Taeyoung Hong (taeyoung.hong@lge.com)

### CODE REVIEW

**Comment by MikyungKim** [comment]
Please resolve the merge conflict.

File: config/webpack.config.js [hunk 1]
```diff
@@ -344,6 +352,15 @@ module.exports = function (env, ilibAdditionalResourcesPath) {
 							}),
 							sideEffects: true
 						},
+						// Adds support for CSS Modules, but using SASS
+						// using the extension .module.scss or .module.sass
+						{
+							test: /\.module\.(scss|sass)$/,
```
Comment (MikyungKim) [inline config/webpack.config.js:358]: Are we supporting only modular Sass/SCSS?

File: package.json [hunk 2]
```diff
@@ -113,6 +113,8 @@
     "react-test-renderer": "^18.0.0",
     "resolution-independence": "^1.0.0",
     "resolve": "^1.22.0",
+    "sass": "^1.49.0",
```
Comment (MikyungKim) [inline package.json:116]: Is this necessary?

File: docs/adding-a-sass-stylesheet.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+---
+
+> Note: this feature is available with `enactjs/cli@5.0.0` and higher.
+
+Generally, we recommend that you don’t reuse the same CSS classes across different components. For example, instead of using a `.Button` CSS class in `<AcceptButton>` and `<RejectButton>` components, we recommend creating a `<Button>` component with its own `.Button` styles, that both `<AcceptButton>` and `<RejectButton>` can render (but [not inherit](https://facebook.github.io/react/docs/composition-vs-inheritance.html)).
+
+Following this rule often makes CSS preprocessors less useful, as features like mixins and nesting are replaced by component composition. You can, however, integrate a CSS preprocessor if you find it valuable.
+
+Thanks to Enact CLI, you can use Sass without installing `sass`.
+Now you can rename `src/App/App.module.less` to `src/App/App.module.scss` and update `src/App/App.js` to import `./App.module.scss`.
```
Comment (MikyungKim) [inline docs/adding-a-sass-stylesheet.md:13]: This sounds like migration from `less` to `scss`. Let's not do this.

**Comment by MikyungKim** [comment]
Please rebase your branch.

File: config/webpack.config.js (hunk 1 above)
Comment (hong6316) [reply config/webpack.config.js:358]: I will adding test: /\.(scss|sass)$/, soon

File: docs/adding-a-sass-stylesheet.md (hunk 3 above)
Comment (hong6316) [reply docs/adding-a-sass-stylesheet.md:13]: OK, I will fix them

File: package.json (hunk 2 above)
Comment (hong6316) [reply package.json:114]: Without this module, errors occurred.

> Failed to compile.
Cannot find module 'sass'
Require stack:
- /home/taeyoung.hong/webos_build_server/WRO-630/cli2/node_modules/sass-loader/dist/utils.js
- /home/taeyoung.hong/webos_build_server/WRO-630/cli2/node_modules/sass-loader/dist/index.js
- /home/taeyoung.hong/webos_build_server/WRO-630/cli2/node_modules/sass-loader/dist/cjs.js
- /home/taeyoung.hong/webos_build_server/WRO-630/cli2/node_modules/loader-runner/lib/loadLoader.js
- /home/taeyoung.hong/webos_build_server/WRO-630/cli2/node_modules/loader-runner/lib/LoaderRunner.js
- /home/taeyoung.hong/webos_build_server/WRO-630/cli2/node_modules/webpack/lib/NormalModuleFactory.js
- /home/taeyoung.hong/webos_build_server/WRO-630/cli2/node_modules/webpack/lib/Compiler.js
- /home/taeyoung.hong/webos_build_server/WRO-630/cli2/node_modules/webpack/lib/webpack.js
- /home/taeyoung.hong/webos_build_server/WRO-630/cli2/node_modules/webpack/lib/index.js
- /home/taeyoung.hong/webos_build_server/WRO-630/cli2/commands/serve.js
- /home/taeyoung.hong/webos_build_server/WRO-630/cli2/bin/enact.js

File: config/webpack.config.js [hunk 4]
```diff
@@ -101,7 +101,7 @@ module.exports = function (env, ilibAdditionalResourcesPath) {
 			{
 				loader: require.resolve('css-loader'),
 				options: Object.assign(
-					{importLoaders: preProcessor ? 2 : 1, sourceMap: shouldUseSourceMap},
+					{sourceMap: shouldUseSourceMap},
```
Comment (MikyungKim) [inline config/webpack.config.js:104]: Is this change necessary?

File: commands/info.js [hunk 5]
```diff
@@ -115,6 +115,7 @@ function api({cliInfo = false, dev = false} = {}) {
 				console.log(`ESLint: ${require('eslint/package.json').version}`);
 				console.log(`Jest: ${require('jest/package.json').version}`);
 				console.log(`LESS: ${require('less/package.json').version}`);
+				console.log(`SASS: ${require('sass/package.json').version}`);
```
Comment (MikyungKim) [inline commands/info.js:118]: I don't think this line is necessary. Please remove it.

File: config/webpack.config.js [hunk 6]
```diff
@@ -337,13 +340,44 @@ module.exports = function (env, ilibAdditionalResourcesPath) {
...
+						// Adds support for CSS Modules, but using SASS
+						// using the extension .module.scss or .module.sass
+						{
+							test: /\.module\.(scss|sass)$/,
+							use: getStyleLoaders(
+								{
+									importLoaders: 3,
+									modules: {
+										mode: 'local',
+										getLocalIdent
```
Comment (MikyungKim) [inline config/webpack.config.js:360]:
```suggestion
										getLocalIdent,
										mode: 'local'

```

Comment (MikyungKim) [reply config/webpack.config.js:360]: For consistency with less

File: package.json (hunk 2 above)
Comment (MikyungKim) [reply package.json:116]: Have you installed `sass` from your app? Please recheck.

File: CHANGELOG.md [hunk 7]
```diff
@@ -4,6 +4,10 @@
 
 * Updated `@enact/template-sandstone` dependency.
 
+### pack, serve
+
+* Added `sass-loader` for supporting sass/scss files.
```
Comment (MikyungKim) [inline CHANGELOG.md:9]:
```suggestion
* Added `sass-loader` for opt-in support for SASS.
```

File: docs/adding-a-sass-stylesheet.md [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+order: 6
+---
+
+> Note: this feature is available with `enactjs/cli@5.0.0` and higher.
+
+Generally, we recommend that you don’t reuse the same CSS classes across different components. For example, instead of using a `.Button` CSS class in `<AcceptButton>` and `<RejectButton>` components, we recommend creating a `<Button>` component with its own `.Button` styles, that both `<AcceptButton>` and `<RejectButton>` can render (but [not inherit](https://facebook.github.io/react/docs/composition-vs-inheritance.html)).
+
+Following this rule often makes CSS preprocessors less useful, as features like mixins and nesting are replaced by component composition. You can, however, integrate a CSS preprocessor if you find it valuable.
+
+Thanks to Enact CLI, you can use Sass without installing `sass`.
```
Comment (MikyungKim) [inline docs/adding-a-sass-stylesheet.md:12]: :duh:
I'll rewrite the documentation.

File: config/webpack.config.js [hunk 9]
```diff
@@ -339,13 +342,45 @@ module.exports = function (env, ilibAdditionalResourcesPath) {
...
 									mode: 'icss'
 								}
 							}),
 							sideEffects: true
 						},
+						// Adds support for CSS Modules, but using SASS
+						// using the extension .module.scss or .module.sass
+						{
+							test: /\.module\.(scss|sass)$/,
+							use: getStyleLoaders(
```
Comment (MikyungKim) [inline config/webpack.config.js:357]: How about creating `getScssStyleLoaders` for simplicity?

File: config/webpack.config.js [hunk 10]
```diff
@@ -339,13 +342,45 @@ module.exports = function (env, ilibAdditionalResourcesPath) {
...
+										mode: 'local'
+
+									}
+								},
+								'sass-loader'
+							)
+						},
+						// Opt-in support for SASS (using .scss or .sass extensions).
+						// By default we support SASS Modules with the
+						// extensions .module.scss or .module.sass
```
Comment (MikyungKim) [inline config/webpack.config.js:371]: These comments are not relevant to us.

**Review by MikyungKim — COMMENTED** [review]
Please add `sass-loader` back.

**Review by MikyungKim — APPROVED** [review]
LGTM
